BambooTracker/tick_counter.cpp

```diff
@@ -58,7 +61,12 @@ void TickCounter::setSpeed(int speed)
 {
 	defStepSize_ = speed;
 	updateTickDifference();
-	tickDiffSum_ = 0;
+	// I choose to not set tickDiffSum_ to 0.
+	// This is because in BambooTracker and FamiTracker,
+	// changing speed does not reset tempo,
+	// so I decided to not interfere with the tempo accumulator either.
+	// However not resetting tickDiffSum_ has the downside that
+	// you can't reset the tempo accumulator mid-song using a speed command.
```

Comment:
Explaining your rationale behind a change is better suited to the commit / PR description than a comment within a function.
```suggestion
```

Reply:
I'm wondering if it's better to *keep* resetting the tempo system when the user inputs a speed effect. The benefit is that it allows the user to reset tempo phase (though you technically can already do so using a groove), which is helpful if they decide to repeatedly switch between tempo 150 and another value (a contrived scenario). The downside is you can't remain at a custom tempo but switch between speed 6 for some rows, and speed 3 for other rows, while maintaining tempo rhythm (also a contrived scenario)... In other words, I don't really know.